### PR TITLE
use as package to export to os via os package

### PR DESCRIPTION
**Pull request by jeffatgametime**
this project supports a CLI entry point to support exporting SSM params to environment variables via shell `export` commands.

this change is to include an entry point for applications to use directly, and export the same SSM params to environment variables using `os.Setenv`.

### CODE REVIEW

File: fetch/fetch.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ssm"
+)
+
+var (
+	client *ssm.SSM
+	paths  []string
+	tags   []string
+
+	trueBool = true
```
Comment (bigodines) [inline fetch/fetch.go:17]: https://github.com/gametimesf/pkg/blob/master/pointer/pointer.go#L9

Comment (ajanata) [reply fetch/fetch.go:17]: -1 this is a public repo since it was forked, it should not depend on stuff in private repos

File: main.go [hunk 2]
```diff
@@ -69,133 +70,21 @@ func initTags(tagsFlag *string) {
...
+	paramVals := make(map[string]string, len(params))
 	for _, param := range params {
 		split := strings.Split(*param.Name, "/")
 		name := split[len(split)-1]
-		fmt.Printf("%s=%s\n", strings.ToUpper(name), *param.Value)
+		paramVals[strings.ToUpper(name)] = *param.Value
 	}
-}
+	return paramVals
+}
```
Comment (ajanata) [inline main.go:90]: no newline at end of file

File: fetch/fetch.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+)
+
+func init() {
+	// initialize AWS client
+	initClient()
+}
+
+func initClient() {
```
Comment (ajanata) [inline fetch/fetch.go:25]: why do this in a separate func?
```suggestion
	// initialize AWS client
```

File: fetch/fetch.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+	if err != nil {
+		panic(err)
+	}
+	nameValues := getParamNameValues(params)
+	for name, value := range nameValues {
+		if err := os.Setenv(name, value); err != nil{
+			panic(err)
+		}
+	}
+}
```
Comment (ajanata) [inline fetch/fetch.go:174]: no newline at end of file

File: fetch/fetch.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+	// initialize AWS client
+	initClient()
+}
+
+func initClient() {
+	session := session.Must(session.NewSession())
+	client = ssm.New(session)
+}
+
+func FetchParams(paths, tags []string) ([]*ssm.Parameter, error) {
```
Comment (ajanata) [inline fetch/fetch.go:30]: I'm assuming there were no actual code changes when moving these from the other file?

File: fetch/fetch.go (hunk 3 above)
Comment (jeffatgametime) [reply fetch/fetch.go:25]: no-look copy pasta

File: fetch/fetch.go (hunk 5 above)
Comment (jeffatgametime) [reply fetch/fetch.go:30]: ah, there were. i'll make sure there's a commit for that.

Comment (jeffatgametime) [reply fetch/fetch.go:30]: actually the first commit shows those changes

File: fetch/fetch.go [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+func MustSetOS(paths, tags []string) {
+	params, err := FetchParams(paths, tags)
+	if err != nil {
+		panic(err)
+	}
+	nameValues := getParamNameValues(params)
+	for name, value := range nameValues {
+		if err := os.Setenv(name, value); err != nil{
```
Comment (ajanata) [inline fetch/fetch.go:165]:
```suggestion
		if err := os.Setenv(name, value); err != nil {
```

**Comment by ajanata** [comment]
> PS: why is this repo public? isn't jamie's original repo already public?

It is impossible to create a private fork. You would have to manually duplicate the repository.

**Comment by bigodines** [comment]
> > PS: why is this repo public? isn't jamie's original repo already public?
> 
> It is impossible to create a private fork. You would have to manually duplicate the repository.

Yes.

**Comment by ajanata** [comment]
You also can't do pull requests from a private repo to a public repo, even if the owner of the public repo can see the private repo...

**Pull request merged**